### DISCUSSION TITLE
Date.setDate() mutates store data

### DISCUSSION THREAD

**Discussion by malvin-rgba** [salesforce/akita#691] [Q&A]
## I'm submitting a...

<pre><code>
[ ] Regression (a behavior that used to work and stopped working in a new release)
[x] Bug report  
[ ] Performance issue
[ ] Feature request
[ ] Documentation issue or request
[ ] Support request
[ ] Other... Please describe:
</code></pre>

## Current behavior

Javascript `Date.setDate()` mutates store

## Expected behavior

It should not be able to mutate readonly store data.

## Minimal reproduction of the problem with instructions

Take a look at `app.component.ts` and `app.state.ts`:
https://stackblitz.com/edit/angular-ivy-ge1q4z?file=src%2Fapp%2Fapp.component.ts

## Questions

- Is this a bug or a known issue?
- Could there potentially be other functions that are able to mutate store data / ignoring immutability principle?
- Should this be documented?

**Comment by NetanelBasal** [comment]
In general, you're the one responsible for not mutate your data. Akita freezes the state in the dev mode, but that's all.

**Reply by malvin-rgba** [reply]
Thank you for the quick reply! For the past, in dev mode, I relied on errors in the console when a store value gets mutated. If there was no such error the assumption was that the application doesn't do so and it will be fine in production. Now I realize that there are edge cases, such as the date, where there is no error thrown in dev mode.

So would you recommend to always clone an object before passing it around through app code just to be sure that nothing mutates that object by functions like setDate or similar?

**Reply by NetanelBasal** [reply]
In such cases, yes.
